### PR TITLE
Merge rc in master

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**

**Description of the Change:**

Merge the RC branch in master ( #3797 #3801)

### CODE REVIEW

**Review by mudit2812 — APPROVED** [review]
Nice

File: pennylane/ops/op_math/evolution.py [hunk 1]
```diff
@@ -93,13 +93,6 @@ def param(self):
         """A real coefficient with ``1j`` factored out."""
         return self.data[0]
 
-    def __repr__(self):
-        return (
-            f"Evolution({self.coeff} {self.base})"
-            if self.base.arithmetic_depth > 0
-            else f"Evolution({self.coeff} {self.base.name})"
-        )
```
Comment (trbromley) [inline pennylane/ops/op_math/evolution.py:101]: Are there changes in this file that we might want to keep for 0.29?
@AlbertMitjans

**Review by trbromley — APPROVED** [review]
Thanks @rmoyard!

File: pennylane/ops/op_math/evolution.py (hunk 1 above)
Comment (AlbertMitjans) [reply pennylane/ops/op_math/evolution.py:101]: We could've kept this `repr` method, but it is not blocking anything.

**Pull request merged**